File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        StringBuilder answer = new StringBuilder();
        int mode = 0;
        for(int i=0;i<code.length();i++){
            if(mode == 0){
                if(code.charAt(i) == '1'){
                    mode = 1;
                    continue;
                }
                if(i%2 == 0)
                    answer.append(code.charAt(i));
                
            } else{
                if(code.charAt(i) == '1'){
                    mode = 0;
                    continue;
                }
                if(i%2 == 1)
                    answer.append(code.charAt(i));
            }
        }
        if(answer.length() == 0)
            answer.append("EMPTY");
        return answer.toString();
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    mode = 0
    for i in range(len(code)):
        if mode == 0:
            if code[i] == '1':
                mode = 1
                continue
            if i%2 == 0:
                answer = answer + code[i];
        else:
            if code[i] == '1':
                mode = 0
                continue
            if i%2 == 1:
                answer = answer + code[i];
    if not answer:
        answer = "EMPTY"
    return answer